packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2470,27 +2470,77 @@ class RenderRepaintBoundary extends RenderProxyBox {
   /// will give you a 1:1 mapping between logical pixels and the output pixels
   /// in the image.
   ///
+  /// ## Sample code
+  ///
+  /// The following is an example of how to go from a `GlobalKey` on a
+  /// `RepaintBoundary` to a PNG:
+  ///
+  /// ```dart
+  /// class PngApp extends StatelessWidget {
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new MaterialApp(
+  ///       title: 'Flutter Demo',
+  ///       theme: new ThemeData(
+  ///         primarySwatch: Colors.blue,
+  ///       ),
+  ///       home: new PngHome(title: 'PNG Demo'),
+  ///     );
+  ///   }
+  /// }
+  ///
+  /// class PngHome extends StatefulWidget {
+  ///   PngHome({Key key, this.title}) : super(key: key);
+  ///
+  ///   final String title;
+  ///
+  ///   @override
+  ///   _PngHomeState createState() => new _PngHomeState();
+  /// }
+  ///
+  /// class _PngHomeState extends State<PngHome> {
+  ///   GlobalKey globalKey = new GlobalKey();
+  ///
+  ///   Future<void> _capturePng() async {
+  ///     RenderRepaintBoundary boundary = globalKey.currentContext.findRenderObject();
+  ///     ui.Image image = await boundary.toImage();
+  ///     ByteData byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+  ///     Uint8List pngBytes = byteData.buffer.asUint8List();
+  ///     print(pngBytes);
```

Comment:
maybe write them to disk rather than to the console? :-P

Reply:
But then they'd go to device disk... and I should really use the `path_provider` plugin, and the example gets dramatically more complicated..

Reply:
Yeah fair enough.